AccountService.js

```diff
@@ -33,7 +33,8 @@ export class AccountService {
       });
       return true;
     } catch(e) {
-      if(e.response && e.response.status && e.response.status === 404) {
+      const {response: {data} = {data: false}} = e;
```

Comment:
Unfortunately, if response is not defined you're going to throw a destructuring error here.

Reply:
tried it in the console and did not throw as long a e is not undefined.

if you don't believe me try it

```js
function strTest(e) {
  const {response: {data} = {data: false}} = e;
  return data;
}
strTest(false);
false
strTest([])
false
strTest({})
false
strTest(undefined)
TypeErrpr: Cannot destructre
strTest(null)
TypeErrpr: Cannot destructure property 'response' of 'undefined' or 'null'
strTest({response: {}})
{}
strTest({response: {type: 'Error'})
{ type: 'Error' }
// we can fix even the undefined and null cases with:
function strTestDefault(e) {
  const {response: {data} = {data: false}} = e || {};
  return data;
}
```

Reply:
major problem with this solution is: _to abstract._

```js
const {data = false} = e.response || {};
```
might be better. I have never seen a catch statement receive null or undefined so I think it is safe.

Reply:
You are correct, I had not grasped the way you were providing a default value for `response`.

wrt to the solution you landed on concerned with the possibility of `e` being undefined...

TIL that we are guaranteed that:
```
All JavaScript and System errors raised by Node.js inherit from, or are instances of, the standard JavaScript <Error> class and are guaranteed to provide at least the properties available on that class.
```
https://nodejs.org/api/errors.html

This is exactly the kind of error handling I was hoping to avoid by using the apisauce wrapper.

Reply:
this is the browser, not node, but yeah I agree so do you like the final solution? if so I can squash all those last few commits into one single solution commit. not sure on how wise it is to rewire dev's history though.

Reply:
I'm comfortable with the latest implementation.  I'll simplify the commit history and merge now.

Reply:
THANKS!